src/target/riscv/riscv_reg.h

```diff
@@ -21,6 +21,16 @@ void riscv_reg_free_all(struct target *target);
 
 /** Write all dirty registers to the target. */
 int riscv_reg_flush_all(struct target *target);
+/**
+ * Checks whether there are dirty registers, reprorting dirty registers usign
+ * the supplied "log_level".
+ */
+bool riscv_reg_any_dirty(const struct target *target, int log_level);
+/**
+ * Invalidate all registers.
```

Comment:
Detail: Minor reword for clarity.

I wanted to stress that this deals with register cache (not with register values in the target).

```suggestion
 * Invalidate all registers - forget their cached register values.
```

Reply:
Addressed.